lib/zinnia.js

```diff
@@ -43,6 +46,22 @@ export async function run ({
   onActivity,
   onMetrics
 }) {
+  const provider = new ethers.JsonRpcProvider(
+    'https://api.node.glif.io/rpc/v1',
+    null,
+    { batchMaxCount: 1 }
+  )
+  const contract = new ethers.Contract(
+    '0xaaef78eaf86dcf34f275288752e892424dda9341',
+    JSON.parse(
+      await fs.readFile(
+        fileURLToPath(new URL('./abi.json', import.meta.url)),
+        'utf8'
+      )
+    ),
+    provider
+  )
```

Comment:
I find it weird that it's the Zinnia module that handles querying the smart contract for scheduled rewards.

In my mind, this functionality is not specific to Zinnia (or Bacalhau or any other module) and, therefore, it belongs to Station Core itself.

What are the arguments in favour of placing this new logic here?

Reply:
This is a good question! My initial reasoning was that bacalhau doesn't use this, so it doesn't belong in `core`, above all modules. And who knows, maybe we will even add another trusted module in the future?

Otoh, it doesn't belong into Zinnia either, because it's the module's decision whether to use Meridian. `ethers` might even run in Zinnia, since for this we don't need to sign messages - or we could just use the jsonrpc endpoint directly - then we only need `fetch`. I think for this we need a way for Zinnia modules to log metrics.

Reply:
Fair enough. Please update our checklist for deploying the new Meridian version to remind us that we need to update this place in Station Core, too?

Reply:
already part of https://github.com/filecoin-station/spark-api/blob/main/docs/impact-evaluator.md :)